src/VisualStudio/Core/Def/Implementation/Progression/ProgressionOptions.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Progression
+{
+    internal static class ProgressionOptions
+    {
+        private const string LocalRegistryPath = @"Roslyn\Internal\OnOff\Components\Progression\";
+
+        public static readonly Option2<bool> SearchUsingNavigateToEngine = new(
+            nameof(ProgressionOptions), nameof(SearchUsingNavigateToEngine), defaultValue: false,
+            storageLocations: new LocalUserProfileStorageLocation(LocalRegistryPath + "SearchUsingNavigateToEngine"));
```

Comment:
I don't see ability to change the value from UI to allow user to turn this on. Is that intentional?

Reply:
Yes.   I don't think we'd do anything to make this user visible.  This would liekly be something we just flip the switch on ourselves.  The purpose here is to allow us to have the optional internally to try out.